packages/components/src/organisms/SKUMatrix/SKUMatrixSidebar.tsx

```diff
@@ -0,0 +1,303 @@
+import Image from 'next/image'
+import React, { useMemo } from 'react'
+import { Badge, Button, QuantitySelector, Skeleton } from '../..'
+import Price, { PriceFormatter } from '../../atoms/Price'
+import Icon from '../../atoms/Icon'
+import { useSKUMatrix, useUI } from '../../hooks'
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+} from '../../molecules/Table'
+import SlideOver, {
+  SlideOverHeader,
+  SlideOverProps,
+} from '../SlideOver'
+
+interface VariationProductColumn {
+  name: string
+  additionalColumns: Array<{ label: string; value: string }>
+  availability: {
+    label: string
+    stockDisplaySettings: 'showStockQuantity' | 'showAvailability'
+  }
+  price: number
+  quantitySelector: number
+}
+
+
+export interface SKUMatrixSidebarProps extends SlideOverProps {
+  /**
+   * Title for the SKUMatrixSidebar component.
+   */
+  title?: string
+  /**
+   * Represents the variations products to building the table.
+   */
+  columns: VariationProductColumn
+
+  /**
+   * Properties related to the 'add to cart' button
+   */
+  buyProps: {
+    'data-testid': string
+    'data-sku': string
+    'data-seller': string
+    onClick(e: React.MouseEvent<HTMLButtonElement>): void
+  }
+  /**
+   * Formatter function that transforms the raw price value and render the result.
+   */
+  formatter?: PriceFormatter
+
+  /**
+   * Check if some result is still loading before render the result.
+   */
+  loading?: boolean
+}
+
+function SKUMatrixSidebar({
+  direction = 'rightSide',
+  title,
+  overlayProps,
+  size = 'partial',
+  children,
+  columns,
+  buyProps: { onClick: buyButtonOnClick, ...buyProps },
+  loading,
+  formatter,
+  ...otherProps
+}: SKUMatrixSidebarProps) {
+  const {
+    setIsOpen,
+    setAllVariantProducts,
+    allVariantProducts,
+    handleChangeQuantityItem,
+  } = useSKUMatrix()
+  const { pushToast } = useUI()
+
+  const cartDetails = useMemo(() => {
+    return allVariantProducts.reduce(
+      (acc, product) => ({
+        amount: acc.amount + product.selectedCount,
+        subtotal: acc.subtotal + product.selectedCount * product.price,
+      }),
+      { amount: 0, subtotal: 0 }
+    )
+  }, [allVariantProducts])
+
+  function resetQuantityItems() {
+    setAllVariantProducts((prev) =>
+      prev.map((item) => ({ ...item, quantity: 0 }))
+    )
+  }
+
+  function handleQuantitySelectorChange(id: string, value: number) {
+    handleChangeQuantityItem(id, value)
+  }
```

Comment:
Do we need this function? can't we call `handleChangeQuantityItem` directly in `onChange` [here](https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci82YTdmYzQ5OTQyMDhjNDFhNzk5MTNkYjBkOWJkNWE5NjcyMTlkMDU0L2YvcGFja2FnZXMvY29tcG9uZW50cy9zcmMvb3JnYW5pc21zL1NLVU1hdHJpeC9TS1VNYXRyaXhTaWRlYmFyLnRzeD91cmw9aHR0cHMlM0ElMkYlMkZnaXRodWIuY29tJTJGY3Vib3MtdnRleCUyRmZhc3RzdG9yZS5naXQ%3D?origin=gitlens)?